api/app_analytics/analytics_db_service.py

```diff
@@ -14,31 +14,103 @@
     FeatureEvaluationBucket,
     Resource,
 )
+from dateutil.relativedelta import relativedelta
 from django.conf import settings
 from django.db.models import Sum
 from django.utils import timezone
 
 from environments.models import Environment
 from features.models import Feature
+from organisations.models import Organisation
 
 ANALYTICS_READ_BUCKET_SIZE = 15
 
 
 def get_usage_data(
-    organisation, environment_id=None, project_id=None
-) -> List[UsageData]:
+    organisation: Organisation,
+    environment_id: int | None = None,
+    project_id: int | None = None,
+    period: str | None = None,
+) -> list[UsageData]:
+    now = timezone.now()
+
+    date_stop = date_start = None
+    period_starts_at = period_ends_at = None
+
+    if period == "current_billing_period":
+        if not getattr(organisation, "subscription_information_cache", None):
+            return []
+        sub_cache = organisation.subscription_information_cache
+        starts_at = sub_cache.current_billing_term_starts_at
+        month_delta = relativedelta(now, starts_at).months
+        period_starts_at = relativedelta(months=month_delta) + starts_at
+        period_ends_at = now
+        date_start = f"-{(now - period_starts_at).days}d"
```

Comment:
Because `date_start` is always relative to the present, we should always use `now` even if we have the same value in `period_ends_at`.

That said, I agree that it looks odd.